### PR TITLE
Switch some mac_unopt tests from intel to arm hosts

### PR DESCRIPTION
**Pull request by zanderso**

### CODE REVIEW

File: testing/run_tests.py [hunk 1]
```diff
@@ -505,7 +505,8 @@ def make_test(name, flags=None, extra_env=None):
     # flutter_desktop_darwin_unittests uses global state that isn't handled
     # correctly by gtest-parallel.
     # https://github.com/flutter/flutter/issues/104789
-    if not os.path.basename(build_dir).startswith('host_debug'):
+    variant = os.path.basename(build_dir)
+    if not variant.startswith('host_debug') and 'arm64' not in variant:
       # Test is disabled for flaking in debug runs:
       # https://github.com/flutter/flutter/issues/127441
       run_engine_executable(
```
Comment (gaaclarke) [inline testing/run_tests.py:512]: So these are effectively not run by CI anymore right?

**Review by gaaclarke — COMMENTED** [review]
Do we want to do this, like is there a prevalence of x64 machines in the pool and moving to arm64 will create a new bottleneck?

File: testing/run_tests.py (hunk 1 above)
Comment (zanderso) [reply testing/run_tests.py:512]: Yup. I don't have time to do this, but the right thing to do would be to split off the desktop unit tests to continue running on an intel machine.

**Comment by zanderso** [comment]
> Do we want to do this, like is there a prevalence of x64 machines in the pool and moving to arm64 will create a new bottleneck?

There are about equal numbers. It might create a bottleneck, but I suspect that long queue times are a better trade-off than flaky tests.

File: testing/run_tests.py (hunk 1 above)
Comment (gaaclarke) [reply testing/run_tests.py:512]: Or make them run on arm64.  Is there an issue for why they can't run on arm64?

Comment (gaaclarke) [reply testing/run_tests.py:512]: I ran them locally on an arm64 machine in debug builds (without gtest-parallel) 5 times and they seem to work.

Comment (gaaclarke) [reply testing/run_tests.py:512]: I reproduced the error by forking this PR (weird the old result wasn't still available): https://github.com/flutter/flutter/issues/157205

Comment (gaaclarke) [reply testing/run_tests.py:512]: I have a fix in review for these tests: https://github.com/flutter/engine/pull/55990

Comment (zanderso) [reply testing/run_tests.py:512]: @gaaclarke I see that PR landed. Is it possible to remove one or both of these guards now?

Comment (gaaclarke) [reply testing/run_tests.py:512]: @zanderso It passed, you are good to rip out that change to run_tests.py

**Comment by zanderso** [comment]
I'm going to shift this to a draft while some internal retries are investigated.

**Review by chinmaygarde — APPROVED** [review]

**Pull request merged**